Files/Helpers/NativeFileOperationsHelper.cs

```diff
@@ -242,29 +242,58 @@ public static bool UnsetFileAttribute(string lpFileName, FileAttributes dwAttrs)
 
         public static string ReadStringFromFile(string filePath)
         {
-            IntPtr hStream = CreateFileFromApp(filePath,
-                GENERIC_READ, 0, IntPtr.Zero, OPEN_EXISTING, (uint)File_Attributes.BackupSemantics, IntPtr.Zero);
-            if (hStream.ToInt64() == -1)
+            IntPtr hFile = CreateFileFromApp(filePath,
+                GENERIC_READ,
+                FILE_SHARE_READ,
+                IntPtr.Zero,
+                OPEN_EXISTING,
+                (uint)File_Attributes.BackupSemantics,
+                IntPtr.Zero);
+
+            if (hFile.ToInt64() == -1)
             {
                 return null;
             }
-            byte[] buff = new byte[4096];
+
+            const int BUFFER_LENGTH = 4096;
+
+            byte[] buffer = new byte[BUFFER_LENGTH];
             int dwBytesRead;
-            string str = null;
+            string szRead = string.Empty;
+
             unsafe
             {
-                fixed (byte* pBuff = buff)
+                bool bRead = false;
+
+                using (MemoryStream msBuffer = new MemoryStream(buffer))
                 {
-                    ReadFile(hStream, pBuff, 4096 - 1, &dwBytesRead, IntPtr.Zero);
-                    //str = Encoding.UTF8.GetString(pBuff, dwBytesRead);
+                    using (StreamReader reader = new StreamReader(msBuffer, true))
+                    {
+                        do
+                        {
+                            fixed (byte* pBuffer = buffer)
+                            {
+                                Array.Clear(buffer, 0, buffer.Length);
```

Comment:
I'm fairly sure we can skip this, we only ever read up to "dwBytesRead" so we ain't gonna read garbage, right?

Reply:
I just left it there to indicate that the array is reused.